pkg/firewalld/firewalld.go

```diff
@@ -0,0 +1,92 @@
+package firewalld
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/godbus/dbus"
+)
+
+const (
+	fInterface = "org.fedoraproject.FirewallD1"
+	fObjPath   = "/org/fedoraproject/FirewallD1"
+)
+
+// Protocol type represents the type of protocol
+type Protocol string
+
+// Supported protocol types
+const (
+	ProtoTCP Protocol = "tcp"
+	ProtoUDP Protocol = "udp"
+)
+
+var (
+	dbusObj   dbus.BusObject
+	isRunning bool
+)
+
+// AddPort removes a port from runtime configuration of firewalld.
+// If zone is set to empty string, it is applied to the default zone.
+func AddPort(zone string, port int, protocol Protocol) error {
+
+	if port <= 0 || (protocol != ProtoTCP && protocol != ProtoUDP) {
```

Comment:
can we do both `max` and `min` port number validation here?

Reply:
It's already done in pmap package which is brick specific. firewalld package is a generic package.